src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,44 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.CustomException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws CustomException {
+        if (user == null) {
+            throw new CustomException("User is null.");
+        }
+        if (!checkLogin(user.getLogin())) {
+            throw new CustomException("The login is null or the length of login is less than 6.");
+        }
+        if (!checkPassword(user.getPassword())) {
+            throw new CustomException("The password is null or the "
+                    + "length of password is less than 6.");
+        }
+        if (!checkAge(user.getAge())) {
+            throw new CustomException("The age is null or is is less than 18.");
+        }
+        User existingUser = storageDao.get(user.getLogin());
+        if (existingUser != null) {
+            throw new CustomException("There is an user with the same login.");
+        }
+        storageDao.add(user);
+        return user;
+    }
+
+    private boolean checkLogin(String login) {
```

Comment:
```suggestion
    private void validateLogin(User user) {
        if (user.getLogin() == null) {
            throw new RegistrationException("Login can't be null.");
        }
        if (user.getLogin().equals("")) {
            throw new RegistrationException("Your login can't be an empty line.");
        }
        if (user.getLogin().length() < MIN_LENGTH) {
            throw new RegistrationException("Your login should contain %s or more symbols."
                    .formatted(MIN_LENGTH));

        }
        if (storageDao.get(user.getLogin()) != null) {....
```

Same for other methods